src/test/java/uk/gov/companieshouse/insolvency/delta/mapper/InsolvencyMapperTest.java

```diff
@@ -0,0 +1,137 @@
+package uk.gov.companieshouse.insolvency.delta.mapper;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.LocalDate;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.util.FileCopyUtils;
+import uk.gov.companieshouse.api.delta.Insolvency;
+import uk.gov.companieshouse.api.delta.InsolvencyDelta;
+import uk.gov.companieshouse.api.insolvency.CaseDates;
+import uk.gov.companieshouse.api.insolvency.InternalCompanyInsolvency;
+import uk.gov.companieshouse.api.insolvency.ModelCase;
+import uk.gov.companieshouse.api.insolvency.PractitionerAddress;
+import uk.gov.companieshouse.api.insolvency.Practitioners;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+        InsolvencyMapperImpl.class,
+        CaseMapperImpl.class,
+        PractitionersMapperImpl.class,
+        PractitionerAddressMapperImpl.class})
+class InsolvencyMapperTest {
+
+    private ObjectMapper mapper;
+    private Insolvency insolvency;
+
+    @Autowired
+    InsolvencyMapper insolvencyMapper;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+        mapper = new ObjectMapper();
+
+        String path = "insolvency-delta-complex.json";
+        String input =
+                FileCopyUtils.copyToString(new InputStreamReader(ClassLoader.getSystemClassLoader().getResourceAsStream(path)));
+
+        final InsolvencyDelta insolvencyDelta = mapper.readValue(input, InsolvencyDelta.class);
+        insolvency = insolvencyDelta.getInsolvency().get(0);
+    }
+
+    @Test
+    void shouldMapInsolvencyToCompanyInsolvency() {
+        String companyNumber = insolvency.getCompanyNumber();
+        InternalCompanyInsolvency insolvencyTarget =
+                insolvencyMapper.insolvencyDeltaToApi(insolvency, companyNumber);
+
+        assertThat(insolvency).isNotNull();
+        assertThat(insolvencyTarget).isNotNull();
+        assertThat(insolvencyTarget.getDeltaAt()).isEqualTo("20211008152823383176");
+        assertThat(insolvencyTarget.getCases().size()).isEqualTo(2);
+
+        // FIRST CASE
+        ModelCase firstCase = insolvencyTarget.getCases().get(0);
+        assertThat(firstCase.getNumber()).isEqualTo(1);
+        assertThat(firstCase.getType()).isEqualTo(ModelCase.TypeEnum.ADMINISTRATIVE_RECEIVER);
+        assertThat(firstCase.getPractitioners().size()).isEqualTo(2);
+
+        // FIRST PRACTITIONER
+        Practitioners firstPractitioner = firstCase.getPractitioners().get(0);
```

Comment:
Instead of manually checking each data point in assertion, probably we can have source.json and expected_target.json files, and we de-serialise both in test, transform source and assert target object is same expected-target?

Reply:
This is a good idea but we have opted for this because at this moment we are not sure the deserialisation works correctly (when we tested that before, we were not using the InternalCompanyInsolvency object) - it is a really comprehensive test case however in the interest of time it was easier to do all the assertions manually without relying on another external for this unit test factor.